### PR TITLE
OneBranch Migration

### PR DESCRIPTION
**Pull request by trwalke**
# OneBranch Migration

- [ ] You've read the [Contributor Guide](https://github.com/AzureAD/microsoft-identity-web/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/AzureAD/microsoft-identity-web/blob/master/CODE_OF_CONDUCT.md).
- [ ] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Updating YAML to enable OneBranch Build.

## Description

SDL tasks such as credscan, policheck, tsa upload, roslyn/codeQL are handled by OneBranch auto injected tasks.

This build only produces signed packages. Tests will come when MSI is working on the pipeline.

Working Legacy build: https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=1293138&view=results
Working OneBranch Build: https://dev.azure.com/IdentityDivision/IDDP/_build/results?buildId=1293136&view=results
Fixes #{bug number} (in this specific format)

IdAbstractionsSourceDir is a variable set on the Legacy and OneBranch builds to specify the source folder.
PipelineType is a variable set on the Legacy and OneBranch builds to specify the type of pipeline. Used to disable tasks on Onebranch

OneBranch Repo PR: https://identitydivision.visualstudio.com/IDDP/_git/Microsoft.Identity.Abstractions%20-%20OneBranch/pullrequest/12648?path=/owners.txt

Fixes #{bug number} (in this specific format)

### CODE REVIEW

File: build/template-pack-and-sign-all-nugets.yaml [hunk 1]
```diff
@@ -10,15 +10,15 @@ steps:
 - template: template-pack-and-sign-nuget.yaml
   parameters:
     BuildConfiguration: ${{ parameters.BuildConfiguration }}
-    ProjectRootPath: '$(Build.SourcesDirectory)\src\Microsoft.Identity.Abstractions'
+    ProjectRootPath: '$(Build.SourcesDirectory)\$(IdAdstractionsSourceDir)src\Microsoft.Identity.Abstractions'
```
Comment (FuPingFranco) [inline build/template-pack-and-sign-all-nugets.yaml:13]:
```suggestion
    ProjectRootPath: '$(Build.SourcesDirectory)\$(IdAbstractionsSourceDir)src\Microsoft.Identity.Abstractions'
```

File: build/template-pack-and-sign-all-nugets.yaml [hunk 2]
```diff
@@ -10,15 +10,15 @@ steps:
...
     AssemblyName: 'Microsoft.Identity.Abstractions'
     HasRefAssembly: 'false'
 
 # Copy all packages out to staging
 - task: CopyFiles@2
-  displayName: 'Copy Files from $(Build.SourcesDirectory) to: $(Build.ArtifactStagingDirectory)\packages'
+  displayName: 'Copy Files from $(Build.SourcesDirectory)\$(IdAdstractionsSourceDir) to: $(Build.ArtifactStagingDirectory)\packages'
```
Comment (FuPingFranco) [inline build/template-pack-and-sign-all-nugets.yaml:19]: Same as above

**Comment by trwalke** [comment]
Updated

**Review by bgavrilMS — APPROVED** [review]

**Review by MZOLN — APPROVED** [review]
lgtm

**Pull request merged**